pyasdf/fits_embed.py

```diff
@@ -0,0 +1,156 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, division, unicode_literals, print_function
+
+"""
+Utilities for embedded ADSF files in FITS.
+"""
+import io
+import re
+
+import numpy as np
+
+from astropy.extern import six
+from astropy.io import fits
+
+from . import asdf
+from . import block
+from . import util
+
+
+ASDF_EXTENSION_NAME = 'ASDF'
+
+FITS_SOURCE_PREFIX = 'fits:'
+
+
+class _FitsBlock(object):
```

Comment:
I'm probably misunderstanding, but is it strictly necessary to have a special kind of "fits block" storage?  In my mind there's nothing special to distinguish a binary block containing a FITS file--and the arrays in the FITS blocks would just be accessed like any other array, with the appropriate offsets.  

That said, I'm not against this either, if you think it makes it clearer what's going on.  There may also be some technical reason I'm missing.